selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/WorkspacesListTest.java

```diff
@@ -0,0 +1,377 @@
+    workspaces.clickOnWorkspaceAddProjectButton(expectedJavaItem.getWorkspaceName());
+
+    workspaceProjects.waitProjectIsPresent(EXPECTED_JAVA_PROJECT_NAME);
+
+    seleniumWebDriver.navigate().back();
+
+    // check "Workspace configuration" button
+    workspaces.waitPageLoading();
+
+    workspaces.moveCursorToWorkspaceRamSection(expectedJavaItem.getWorkspaceName());
+    workspaces.clickOnWorkspaceConfigureButton(expectedJavaItem.getWorkspaceName());
+
+    workspaceConfig.waitConfigForm();
+
+    assertEquals(
+        workspaceConfig.createExpectedWorkspaceConfig(expectedJavaItem.getWorkspaceName()),
+        workspaceConfig.getWorkspaceConfig());
+
+    seleniumWebDriver.navigate().back();
+
+    // check stop/start button
+    workspaces.waitPageLoading();
+
+    workspaces.moveCursorToWorkspaceRamSection(expectedJavaItem.getWorkspaceName());
+    workspaces.clickOnWorkspaceStopStartButton(expectedJavaItem.getWorkspaceName());
+    workspaces.waitWorkspaceStatus(expectedJavaItem.getWorkspaceName(), Statuses.STOPPED);
+
+    workspaces.clickOnWorkspaceStopStartButton(expectedJavaItem.getWorkspaceName());
+    workspaces.waitWorkspaceStatus(expectedJavaItem.getWorkspaceName(), Statuses.RUNNING);
+
+    // check adding the workspace to list
+    workspaces.clickOnAddWorkspaceBtn();
+    newWorkspace.waitToolbar();
+    newWorkspace.typeWorkspaceName(NEWEST_CREATED_WORKSPACE_NAME);
+    newWorkspace.clickOnCreateButtonAndEditWorkspace();
+    workspaceOverview.checkNameWorkspace(NEWEST_CREATED_WORKSPACE_NAME);
+
+    assertEquals(dashboard.getWorkspacesCountInWorkspacesItem(), EXPECTED_WORKSPACES_COUNT + 1);
+
+    dashboard.selectWorkspacesItemOnDashboard();
+
+    workspaces.waitPageLoading();
+    assertEquals(workspaces.getVisibleWorkspacesCount(), EXPECTED_WORKSPACES_COUNT + 1);
+
+    Workspaces.WorkspaceListItem newestCreatedWorkspaceItem =
+        workspaces.getWorkspacesListItemByWorkspaceName(
+            workspaces.getVisibleWorkspaces(), NEWEST_CREATED_WORKSPACE_NAME);
+
+    assertEquals(newestCreatedWorkspaceItem, expectedNewestWorkspaceItem);
+
+    // delete workspaces by checkboxes
+    workspaces.selectWorkspaceByCheckbox(expectedNewestWorkspaceItem.getWorkspaceName());
+    workspaces.clickOnDeleteWorkspacesBtn();
+    workspaces.clickOnDeleteButtonInDialogWindow();
+
+    workspaces.waitWorkspaceIsNotPresent(expectedNewestWorkspaceItem.getWorkspaceName());
+
+    workspaces.selectAllWorkspacesByBulk();
+    workspaces.clickOnDeleteWorkspacesBtn();
+    workspaces.clickOnDeleteButtonInDialogWindow();
+
+    workspaces.waitWorkspaceIsNotPresent(expectedBlankItem.getWorkspaceName());
+    workspaces.waitWorkspaceIsNotPresent(expectedJavaItem.getWorkspaceName());
+  }
+
+  private void deleteIfWorkspaceExists(String workspaceName) throws Exception {
+    if (isWorkspaceExist(workspaceName)) {
+      testWorkspaceServiceClient.delete(workspaceName, defaultTestUser.getName());
```

Comment:
moreover - we doesn't need to check if workspace exists or not because `testWorkspaceServiceClient.delete()` has already done it at start.